### PR TITLE
add controller, mapper and repository in imageProjects

### PR DESCRIPTION
**Pull request by VLiamin**

### CODE REVIEW

File: src/ImageService.Business/Commands/ImageProject/GetImageProjectCommand.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+        public OperationResultResponse<ImageDataResponse> Execute(Guid parentId)
+        {
+            OperationResultResponse<ImageDataResponse> response = new();
+
+            DbImagesProject dbImageProject = _imageProjectRepository.Get(parentId);
+
+            if (dbImageProject == null)
+            {
+                response.Body = null;
```
Comment (Cheephot) [inline src/ImageService.Business/Commands/ImageProject/GetImageProjectCommand.cs:33]: Add response.Status = OperationResultStatusType.Failed;

**Review by Cheephot — APPROVED** [review]

File: src/ImageService.Business/Commands/ImageProject/GetImageProjectCommand.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+using LT.DigitalOffice.Kernel.Responses;
+using LT.DigitalOffice.Models.Broker.Responses.File;
+using System;
+
+namespace LT.DigitalOffice.ImageService.Business.Commands.ImageProject
+{
+    public class GetImageProjectCommand : IGetImageProjectCommand
+    {
+        private readonly IImageProjectRepository _imageProjectRepository;
+        private readonly IImageResponseMapper _imageResponseMapper;
```
Comment (MariyaSin) [inline src/ImageService.Business/Commands/ImageProject/GetImageProjectCommand.cs:16]: add empty line

File: src/ImageService.Business/Commands/ImageProject/GetImageProjectCommand.cs (hunk 1 above)
Comment (VLiamin) [reply src/ImageService.Business/Commands/ImageProject/GetImageProjectCommand.cs:33]: done

File: src/ImageService.Business/Commands/ImageProject/GetImageProjectCommand.cs (hunk 2 above)
Comment (VLiamin) [reply src/ImageService.Business/Commands/ImageProject/GetImageProjectCommand.cs:16]: done

File: src/ImageService.Data/ImageProjectRepository.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+namespace LT.DigitalOffice.ImageService.Data
+{
+    public class ImageProjectRepository : IImageProjectRepository
+    {
+        private readonly IDataProvider _provider;
+
+        public ImageProjectRepository(IDataProvider provider)
+        {
+            this._provider = provider;
```
Comment (MariyaSin) [inline src/ImageService.Data/ImageProjectRepository.cs:18]: no need write "this"

File: src/ImageService.Data/ImageProjectRepository.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+            _provider.ImagesProjects.AddRange(imagesProjects);
+            _provider.Save();
+
+            return imagesProjects.Select(x => x.Id).ToList();
+        }
+
+        public bool Delete(Guid imageId)
+        {
+            throw new NotImplementedException();
```
Comment (MariyaSin) [inline src/ImageService.Data/ImageProjectRepository.cs:36]: ??? where is the implementation?

File: src/ImageService.Data/ImageProjectRepository.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        }
+
+        public List<DbImagesProject> Get(List<Guid> imageIds)
+        {
+            return _provider.ImagesProjects.Where(x => imageIds.Contains(x.Id)).ToList();
+        }
+
+        public DbImagesProject Get(Guid imageId)
+        {
+            return _provider.ImagesProjects.Where(x => x.Id == imageId).FirstOrDefault();
```
Comment (MariyaSin) [inline src/ImageService.Data/ImageProjectRepository.cs:46]: What for do you use "Where"?

File: src/ImageService.Mappers/Db/DbImagesProjectMapper.cs [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+                Id = Guid.NewGuid(),
+                ParentId = parentId,
+                Name = name,
+                Content = content,
+                Extension = extension,
+                CreatedAtUtc = DateTime.UtcNow,
+                CreatedBy = createdBy
+            };
+        }
+
```
Comment (MariyaSin) [inline src/ImageService.Mappers/Db/DbImagesProjectMapper.cs:26]: remove line

File: src/ImageService.Data/ImageProjectRepository.cs (hunk 4 above)
Comment (VLiamin) [reply src/ImageService.Data/ImageProjectRepository.cs]: done

File: src/ImageService.Data/ImageProjectRepository.cs (hunk 5 above)
Comment (VLiamin) [reply src/ImageService.Data/ImageProjectRepository.cs]: done

File: src/ImageService.Data/ImageProjectRepository.cs (hunk 3 above)
Comment (VLiamin) [reply src/ImageService.Data/ImageProjectRepository.cs]: done

File: src/ImageService.Mappers/Db/DbImagesProjectMapper.cs (hunk 6 above)
Comment (VLiamin) [reply src/ImageService.Mappers/Db/DbImagesProjectMapper.cs]: done

File: LT.DigitalOffice.ImageService.sln [hunk 7]
```diff
@@ -9,7 +9,7 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solutions Items", "Solution
 	ProjectSection(SolutionItems) = preProject
 		..\.dockerignore = ..\.dockerignore
 		..\.gitignore = ..\.gitignore
-		..\Common.Configurations.targets = ..\Common.Configurations.targets
+		Common.Configurations.targets = Common.Configurations.targets
```
Comment (BelehovEgor) [inline LT.DigitalOffice.ImageService.sln:12]: is it correct?

File: src/ImageService.Mappers/Db/Interfaces/IDbImageProjectMapper.cs [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.ImageService.Mappers.Db.Interfaces
+{
+    [AutoInject]
+    public interface IDbImageProjectMapper
+    {
+        DbImagesProject Map(Guid parentId, string name, string content, string extension, Guid createdBy);
```
Comment (BelehovEgor) [inline src/ImageService.Mappers/Db/Interfaces/IDbImageProjectMapper.cs:14]: maybe better DbImageProject?

File: src/ImageService.Mappers/Models/ImageResponseMapper.cs [hunk 9]
```diff
@@ -0,0 +1,20 @@
+﻿using LT.DigitalOffice.ImageService.Mappers.Models.Interfaces;
+using LT.DigitalOffice.ImageService.Models.Db;
+using LT.DigitalOffice.ImageService.Models.Dto.Responses;
+
+namespace LT.DigitalOffice.ImageService.Mappers.Models
+{
+    public class ImageResponseMapper : IImageResponseMapper
+    {
+        public ImageDataResponse  Map(DbImagesProject dbImagesProject)
```
Comment (BelehovEgor) [inline src/ImageService.Mappers/Models/ImageResponseMapper.cs:9]: check  dbImagesProject != null

File: src/ImageService.Models.Dto/Responses/ImageDataResponse .cs [hunk 10]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace LT.DigitalOffice.ImageService.Models.Dto.Responses
+{
+    public class ImageDataResponse
```
Comment (BelehovEgor) [inline src/ImageService.Models.Dto/Responses/ImageDataResponse .cs:5]: record, all models in dto project are records or enum

File: src/ImageService.Mappers/Db/DbImageProjectMapper.cs [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.ImageService.Mappers.Db
+{
+    public class DbImageProjectMapper : IDbImageProjectMapper
+    {
+        public DbImagesProject Map(Guid parentId, string name, string content, string extension, Guid createdBy)
```
Comment (BelehovEgor) [inline src/ImageService.Mappers/Db/DbImageProjectMapper.cs:13]: same comment

File: LT.DigitalOffice.ImageService.sln (hunk 7 above)
Comment (VLiamin) [reply LT.DigitalOffice.ImageService.sln]: done

File: src/ImageService.Mappers/Models/ImageResponseMapper.cs (hunk 9 above)
Comment (VLiamin) [reply src/ImageService.Mappers/Models/ImageResponseMapper.cs]: done

File: src/ImageService.Mappers/Db/Interfaces/IDbImageProjectMapper.cs (hunk 8 above)
Comment (ndoriomedov) [reply src/ImageService.Mappers/Db/Interfaces/IDbImageProjectMapper.cs:10]: will do it in other task

File: src/ImageService.Models.Dto/Responses/ImageDataResponse .cs (hunk 10 above)
Comment (VLiamin) [reply src/ImageService.Models.Dto/Responses/ImageDataResponse .cs]: done

File: src/ImageService.Data/ImageProjectRepository.cs [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+        public ImageProjectRepository(IDataProvider provider)
+        {
+            _provider = provider;
+        }
+
+        public List<Guid> Create(List<DbImagesProject> imagesProjects)
+        {
+            if (imagesProjects.Contains(null))
+            {
+                throw new ArgumentNullException(nameof(imagesProjects));
```
Comment (MariyaSin) [inline src/ImageService.Data/ImageProjectRepository.cs:23]: do not throw exception, return null

File: src/ImageService.Data/ImageProjectRepository.cs [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+            }
+
+            _provider.ImagesProjects.AddRange(imagesProjects);
+            _provider.Save();
+
+            return imagesProjects.Select(x => x.Id).ToList();
+        }
+
+        public bool Delete(List<DbImagesProject> imagesProjects)
```
Comment (MariyaSin) [inline src/ImageService.Data/ImageProjectRepository.cs:32]: why do you not accept List of ids? you can get dbImagesProject right here, by ids.

File: src/ImageService.Data/ImageProjectRepository.cs [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+        }
+
+        public List<DbImagesProject> Get(List<Guid> imageIds)
+        {
+            return _provider.ImagesProjects.Where(x => imageIds.Contains(x.Id)).ToList();
+        }
+
+        public DbImagesProject Get(Guid imageId)
+        {
+            return _provider.ImagesProjects.Find(imageId);
```
Comment (MariyaSin) [inline src/ImageService.Data/ImageProjectRepository.cs:52]: FirstOrDefault?

File: src/ImageService.Mappers/Models/ImageResponseMapper.cs [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+                return new ImageDataResponse
+                {
+                    Id = dbImageProject.Id,
+                    Content = dbImageProject.Content,
+                    Name = dbImageProject.Name,
+                    Extension = dbImageProject.Extension
+                };
+            }
+
+            throw new ArgumentNullException(nameof(dbImageProject));
```
Comment (MariyaSin) [inline src/ImageService.Mappers/Models/ImageResponseMapper.cs:23]: please, do not throw exception here))

File: src/ImageService.Data/ImageProjectRepository.cs (hunk 14 above)
Comment (VLiamin) [reply src/ImageService.Data/ImageProjectRepository.cs:52]: Find returns single element. Why do we need FirstOrDefault?

File: src/ImageService.Mappers/Models/ImageResponseMapper.cs (hunk 15 above)
Comment (VLiamin) [reply src/ImageService.Mappers/Models/ImageResponseMapper.cs]: done

File: src/ImageService.Data/ImageProjectRepository.cs (hunk 12 above)
Comment (VLiamin) [reply src/ImageService.Data/ImageProjectRepository.cs]: done

File: src/ImageService.Data/ImageProjectRepository.cs (hunk 13 above)
Comment (VLiamin) [reply src/ImageService.Data/ImageProjectRepository.cs]: done

File: src/ImageService.Data/ImageProjectRepository.cs [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+        }
+
+        public bool Delete(List<Guid> imageIds)
+        {
+            if (imageIds == null)
+            {
+                return false;
+            }
+
+            _provider.ImagesProjects.RemoveRange(Get(imageIds));
```
Comment (MariyaSin) [inline src/ImageService.Data/ImageProjectRepository.cs:39]: you remove the only image by id, you must remove parentImage too

**Pull request closed without merging**